make/variables.sh

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
```

Comment:
as discussed `validate_input` should be invoked with variables.sh

Reply:
to be fair validate probably should reside with variables.sh... thinking out loud

Reply:
I have moved it to `variables.sh`

The disadvantage was that you could no longer invoke it as a separate `makefile` target for pre-run validation without the intent to start deploying, but mitigated that by invoking `variables.sh` in the `init` target.